ksqldb-functional-tests/src/test/resources/query-validation-tests/avro.json

```diff
@@ -667,6 +667,33 @@
         {"topic": "OUTPUT", "value": null}
       ]
     },
+    {
+      "name": "BOOLEAN - key - inference",
+      "properties": {
+        "ksql.key.format.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM INPUT (foo INT) WITH (kafka_topic='input_topic', format='AVRO');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM INPUT;"
+      ],
+      "topics": [
+        {
+          "name": "input_topic",
+          "keySchema": {"type": "boolean"},
+          "keyFormat": "AVRO",
+          "valueSchema": {"name": "ignored", "type": "record", "fields": [{"name": "FOO", "type": "int"}]},
```

Comment:
Is there a purpose to including this value schema here, given that the test does not use value inference? If we want to test that this value schema is ignored in favor of the schema that's explicitly supplied in the statement, we should update this schema to be different. If not, can we remove this in order to clean up the tests?

Or does the QTT framework require that it's present for Avro types, even if the schema inference isn't used...?

Reply:
I don't think it's necessary - I can add separate tests to make sure that it's respected if the schema differs.

Reply:
Actually, it looks like QTT framework does require this even if it's present for Avro types. We can look at removing that later

Reply:
OK - I figured out what's going on and I'll fix it in a follow-up PR. We generate the topic information from the CT/CS statement if there is no topic node in the QTT. The problem is that if we have either the key or the value schema we need both because otherwise it won't generate the topic from the CT statement and use the one supplied (and we get a 404 when looking up the schema in SR). The fix isn't necessarily straightforward, so for now whenever we are using inference I'll just supply the key and value schema.

Reply:
https://github.com/confluentinc/ksql/issues/6441#issue-722459755

Reply:
Thanks for investigating! Agreed we shouldn't block this PR.